open-apparel-registry/open-apparel-registry
# Add facility_location to FacilityClaim to allow for prioritizing claim address/location in facility details

## Overview

Add facility_location to FacilityClaim to allow for prioritizing claim address/location in facility details

Connects #1754

## Demo

<img width="355" alt="Screen Shot 2022-08-03 at 9 55 41 AM" src="https://user-images.githubusercontent.com/21046714/182629404-20885b83-b53a-4b43-bbb9-37ac3f66c8bf.png">
<img width="349" alt="Screen Shot 2022-08-03 at 10 18 40 AM" src="https://user-images.githubusercontent.com/21046714/182632314-b8250d50-8284-4c02-9307-7d5308f94306.png">

## Testing Instructions

On `develop`:
* As c2@example.com, upload and process [TestGeoCoding.csv](https://github.com/open-apparel-registry/open-apparel-registry/files/9252095/TestGeoCoding.csv). Four of the facilities have real addresses and claim addresses. One (marked 'fake') has a very incorrect claim address, 
* Claim each facility. 
* As c1@example.com, approve each facility.
* As c2@example.com, assign an address to each facility and save. Use the addresses listed as `claim_address` in TestGeoCoding.csv. One address is left intentionally blank to ensure migrating claims without addresses will succeed. 

On this branch:
* Run `./scripts/manage migrate`
* Run `./scripts/manage set_facility_locations_from_claims`
     * You should see one failed facility. The other facilities should succeed. The failure should not prevent the command from completing. 
* Using the 'My Facilities' page, visit each claimed facility in turn. 
     * The claim address and claim location (under 'GPS') should be listed and prioritized, except for Test 5 (fake) and Test 2, where the original facility location should be listed instead. 
     * If the claim location and original facility location are significantly different, you should see both listed; otherwise, only the claim location will be listed. 
     * If the claim address and the original facility address are different, you should see both listed; otherwise, only the claim address will be listed.
* Go to the claim update page for Test 2. Add '990 Spring Garden St, Philadelphia, PA' as the address and save. You should see a failure message. Add a valid address in China and save; it should succeed. 
     * The facility details for Test 2 should now show the new claim location. 
* Clear the address from Test 2 in the claim update page, and save. It should succeed.
     * The facility details for Test 2 should now show the original claim location. 

## Checklist
